[14.0][MIG] account_reconcile_model_strict_match_amount: Migration to 14.0

nguyenhk99 opened the pull request.

grindtildeath committed:
[ADD] account_reconcile_model_strict_match_amount

grindtildeath committed:
[IMP] account_reconcile_model_strict_match_amount: black, isort, prettier

grindtildeath committed:
[MIG] account_reconcile_model_strict_match_amount: Migration to 13.0

Commit:
[IMP] account_reconcile_model_strict_match_amount: black, isort, prettier

Commit:
[MIG] account_reconcile_model_strict_match_amount: Migration to 14.0

nilshamerlinck commented:
Ideally https://github.com/OCA/account-reconcile/pull/411 would need to be merged before this one

grindtildeath commented on account_reconcile_model_strict_match_amount/tests/test_account_reconcile_model_strict_match_amount.py:
```diff
@@ -105,26 +117,26 @@ def test_auto_reconcile_strict_match_90(self):
+                self.bank_line_s_3.id: {
+                    "aml_ids": [self.invoice_line_s_5.id],
                     "model": my_rule,
-                    "status": "reconciled",
```
This flag shouldn't be removed as it defines if a statement line is reconciled.

grindtildeath commented on account_reconcile_model_strict_match_amount/tests/test_account_reconcile_model_strict_match_amount.py:
```diff
@@ -9,57 +9,69 @@ class TestAccountReconcileModelStrictMatchAmount(TestReconciliationMatchingRules
+                "statement_id": cls.bank_st_s.id,
                 "name": "ABC001XYZ",
-                "partner_id": cls.partner_3.id,
+                "payment_ref": "REF0003",
```
As I understand, the label in v14.0 is the `payment_ref` field, so we need to use the invoice reference in there.

grindtildeath committed:
account_reconcile_model_strict_match_amount: Fix tests for v14.0

grindtildeath reviewed (changes requested):
Looks good but the tests need fixing. I did it for you here: https://github.com/HoKhanhNguyen99/account-reconcile/pull/1 :wink:

grindtildeath commented on account_reconcile_model_strict_match_amount/models/account_reconcile_model.py:
```diff
@@ -0,0 +1,79 @@
+                END >= {match_total_amount_param}
+            """.format(
+                match_total_amount_param=self.match_total_amount_param
+            )
```
This also need to be rewritten in consideration of the changes in v14.0, because it wouldn't take into account the new setting `match_text_location_label` on the reconciliation model.

nguyenhk99 replied:
> Looks good but the tests need fixing. I did it for you here: [HoKhanhNguyen99#1](https://github.com/HoKhanhNguyen99/account-reconcile/pull/1) wink

Thank @grindtildeath for your review. I just push up.

Commit:
account_reconcile_model_strict_match_amount: Fix _get_select_communication_flag and _get_select_payment_reference_flag

nguyenhk99 commented:
Hi @grindtildeath, all your feedbacks have been taken into account, would you mind reviewing this PR again?

leemannd commented on account_reconcile_model_strict_match_amount/models/account_reconcile_model.py:
```diff
@@ -14,66 +14,50 @@ class AccountReconcileModel(models.Model):
-                END >= {match_total_amount_param}
-            """.format(
+            return comm_flag
+        # comm_flag.replace is for 13.0
```
I'm not sure to understand this comment

hailangvn replied:
I think the comment regards https://github.com/OCA/account-reconcile/pull/410/commits/70cf821a4f0526b6320f361a3f6a17a8fb93fd6a#diff-561c565bcdc3ab79d666e97f6dae8f5aee3051c29282c5f0c74c0483d3468a3bR26, which is intended to be ported to 13.0 if needed.

grindtildeath reviewed (approved):
LGTM :+1:

grindtildeath committed:
Fix account_reconcile_model_strict_match_amount tests

Tests were broken after introduction of new commit on Odoo.
Since account.move.line.ref field was set as readonly, writing
a value on it did not set the value on account.move.ref (which
it's related to) anymore.
cf https://github.com/odoo/odoo/commit/38c13bbef39e3e82a77cfd4a316be94280e3cee5

grindtildeath commented:
@HoKhanhNguyen99 Can you please rebase and check if we don't need to apply the same change as #456 ?

nguyenhk99 commented:
> @HoKhanhNguyen99 Can you please rebase and check if we don't need to apply the same change as #456 ?

Hi @grindtildeath , I just force push, please check it out. Thank you.

dreispt commented:
/ocabot merge nobump

grindtildeath commented on account_reconcile_model_strict_match_amount/tests/test_account_reconcile_model_strict_match_amount.py:
```diff
@@ -0,0 +1,147 @@
+)
+
+
+class TestAccountReconcileModelStrictMatchAmount(TestReconciliationMatchingRules):
```
```suggestion
 @tagged("post_install", "-at_install")
class TestAccountReconcileModelStrictMatchAmount(TestReconciliationMatchingRules):
```

grindtildeath commented on account_reconcile_model_strict_match_amount/tests/test_account_reconcile_model_strict_match_amount.py:
```diff
@@ -0,0 +1,147 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo.addons.account.tests.test_reconciliation_matching_rules import (
```
```suggestion
from odoo.tests import tagged

from odoo.addons.account.tests.test_reconciliation_matching_rules import (
```

grindtildeath commented:
@HoKhanhNguyen99 Can you please commit my suggestions and we'll try to merge again?

nguyenhk99 commented:
> @HoKhanhNguyen99 Can you please commit my suggestions and we'll try to merge again?

Hi @grindtildeath , the PR has been updated. Please check again. Thanks.

grindtildeath commented:
/ocabot merge nobump
